transaction/DirectedGraph.hpp

```diff
@@ -0,0 +1,173 @@
+/**
+ *   Copyright 2016, Quickstep Research Group, Computer Sciences Department,
+ *     University of Wisconsin—Madison.
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License");
+ *   you may not use this file except in compliance with the License.
+ *   You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   Unless required by applicable law or agreed to in writing, software
+ *   distributed under the License is distributed on an "AS IS" BASIS,
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *   See the License for the specific language governing permissions and
+ *   limitations under the License.
+ **/
+
+#ifndef QUICKSTEP_TRANSACTION_DIRECTED_GRAPH_HPP_
+#define QUICKSTEP_TRANSACTION_DIRECTED_GRAPH_HPP_
+
+#include <algorithm>
+#include <memory>
+#include <unordered_map>
+#include <unordered_set>
+#include <stack>
+#include <vector>
```

Comment:
The includes should be alphabetically sorted. I would highly recommend doing it through an IDE instead of sorting it manually. e.g. in vim, you can select a visual block and issue a sort command. You may want to find something similar for emacs.